src/panel-layout/interfaces.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { ReactNode } from 'react';
+
+import { BaseComponentProps } from '../internal/base-component';
+import { NonCancelableEventHandler } from '../internal/events';
+
+export interface PanelLayoutProps extends BaseComponentProps {
+  /**
+   * Position of the panel with respect to the main content
+   */
+  panelPosition?: PanelLayoutProps.PanelPosition;
+
+  /**
+   * Initial panel size, for uncontrolled behavior.
+   *
+   * The actual size may vary, depending on `minPanelSize` and `maxPanelSize`.
+   */
+  defaultPanelSize?: number;
+
+  /**
+   * Size of the panel. If provided, and panel is resizable, the component is controlled,
+   * so you must also provide `onPanelResize`.
+   *
+   * The actual size may vary, depending on `minPanelSize` and `maxPanelSize`.
+   */
+  panelSize?: number;
+
+  /**
+   * The minimum size of the panel.
```

Comment:
Maybe we can state it here as: "when provided panelSize is less than minPanelSize the minPanelSize will be used instead" (or alike).

Reply:
Updated the wording within `panelSize`/`defaultPanelSize` to make this clearer